### PR TITLE
v1.7.0

### PR DESCRIPTION
**Pull request by C-D-Lewis**
_Features_
- Use evrythng.js@5.4.0
- Add `postinstall` message to help getting started
- Always switch to a newly added `operator`
- Add `version`, `requireVersion()`, and `getOperatorScope()` to the plugin API.
- Show the currently selected operator in the help text (`evrythng`)

_Fixes_
- Fix `npm audit` warnings
- Reorder help text to better show installed plugins
- `redirections` always use the supplied short domain, instead of assuming the default.

_TODO_
- [ ] Bump version after review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 > Requires Node.js version 7.6 or greater
 
-Command Line Interface (CLI) for working with the
```
Comment (iamfrontender) [inline README.md]: I was always looking at those hanging spaces and thinking, 'What a mess'. I feel so much better now.

File: README.md [hunk 2]
```diff
@@ -109,17 +113,17 @@ module.exports = (api) => {
     },
   };
 
-  // Register a new command
+  // Register the new command
```
Comment (iamfrontender) [inline README.md]: :police_car:

File: package.json [hunk 3]
```diff
@@ -1,9 +1,11 @@
 {
   "name": "evrythng-cli",
   "version": "1.6.0",
-  "description": "CLI for the EVRYTHNG API.",
+  "description": "Extensible command line interface for the EVRYTHNG API.",
```
Comment (iamfrontender) [inline package.json]: Next version would be: `Extensible command line interface for EVRYTHNG`

File: src/modules/api.js [hunk 4]
```diff
@@ -70,10 +61,74 @@ const API = {
...
+  },
+
+  /**
+   * Require a specific minimum CLI version to run.
+   *
+   * @params {string} Semver version to require, e.g: '1.7.0'.
+   * @throws If the version required is less than the current running CLI version.
+   */
+  requireVersion: (spec) => {
+    if (semverCompare(version, spec) < 0) {
```
Comment (iamfrontender) [inline src/modules/api.js]: Should it also throw if plugin requires incompatible semver, for example `require: '^7.0.0'` should throw if CLI is `6.2.1` or `8.2.2`. 
Possible to implement with `semver` module, `satisfies`, as:
```
semver.satisfies(currentVersion, requiredByPlugin)
```
Where required by plugin could be a semver expression, such as:
```
1.x || >=2.5.0 || 5.0.0 - 7.2.3
```

File: src/modules/config.js [hunk 5]
```diff
@@ -22,16 +22,13 @@ const DEFAULT_CONFIG = {
 };
 
 const CONFIG_SCHEMA = {
-  type: 'object',
```
Comment (iamfrontender) [inline src/modules/config.js]: JSON schema is so convoluted... is it object by default?

File: src/commands/product.js [hunk 6]
```diff
@@ -108,20 +108,37 @@ module.exports = {
...
+      pattern: '$id redirections $shortDomain create $payload',
     },
-    readProductRedirection: {
-      execute: async ([id]) => http.get(`/products/${id}/redirector`),
-      pattern: '$id redirection read',
+    listProductRedirections: {
+      execute: async ([id, , shortDomain]) => http.shortDomainRequest(shortDomain, {
+        params: { evrythngId: id },
+      }),
+      pattern: '$id redirections $shortDomain list',
```
[the remark this answers is not in the document]
Comment (iamfrontender) [reply src/commands/product.js]: Meant, from all of the short domains

File: src/modules/config.js (hunk 5 above)
Comment (C-D-Lewis) [reply src/modules/config.js]: It's implied if you specify some `properties`

File: src/modules/api.js [hunk 7]
```diff
@@ -70,10 +61,74 @@ const API = {
...
   getConfig: () => config,
-  runCommand,
+
+  /**
+   * Run an existing CLI command, as a list of tokens.
+   *
+   * @param {string[]} args - The args to identify and run.
+   * @returns {Promise<object>} The command result, such as API response data.
+   */
+  runCommand: async (args) => {
```
[the remark this answers is not in the document]
Comment (C-D-Lewis) [reply src/modules/api.js]: I did try this, but it explodes when it gets to JSON:
`evrythng thngs create '{"name": "Test Thng"}`
becomes
`[ 'evrythng', 'thngs', 'create', '\'{"name":', '"Test', 'Thng"}\'' ]`

File: src/commands/product.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (C-D-Lewis) [reply src/commands/product.js]: This does return an array of all? Or do you mean for all short domains?

File: src/modules/api.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (iamfrontender) [reply src/modules/api.js]: Indeed, there's no easy way to make this more convenient

File: src/commands/product.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (C-D-Lewis) [reply src/commands/product.js]: In the future!

File: src/modules/api.js (hunk 4 above)
Comment (C-D-Lewis) [reply src/modules/api.js]: Done

**Pull request merged**